### ISSUE TITLE
DNS reconfig fails when using systemd implementation of resolvconf

### ISSUE THREAD

**Issue by danderson** [tailscale/tailscale#631]
During bringup on my NixOS system, the DNS config stack detects systemd's implementation of resolvconf (because systemd is in the per-service $PATH generated by NixOS, even though _really_ my system uses openresolv - that's a bug in the NixOS module that I'll fix separately).

However, DNS config fails with:

router: dns up: resolvconf: running /nix/store/7avxs5gsi32ci1v8qpvr3sqc15fnx90q-systemd-243.7/bin/resolvconf -m 0 -x -a tun-tailscale.inet: Unknown interface 'tun-tailscale': No such device

Looks like systemd's shim implementation doesn't like us providing a configuration for a fake interface name :+1: 

cc @dshynkev

**Comment by dshynkev** [comment]
This also tells me that my heuristic for detecting systemd is imperfect, as `resolvconf` should never get used when it's the systemd shim. This is probably because `systemctl` is not in the PATH, which makes sense on NixOS (and it). I wonder: why is `resolvconf` in the PATH? Is this something you specified in the NixOS unit?

In any case, potentially better heuristics are:
1. Just check `/etc/resolv.conf` for `127.0.0.53`, which is probably reasonably unique for systemd (can I read that file on NixOS? I assume so, for compatibility reasons, right?)
2. Try to talk to systemd-resolved over DBus and see if I fail. This is probably the most "correct" way to discover it, but I don't immediately see a good way to discover whether resolved is _managing_ DNS or merely serving as a client. On the other hand, we may want to _always_ push our config to resolved as well as whatever else is active; I think it's harmless, if not useful.

**Comment by orangeturtle739** [comment]
Is there a workaround for this? I'm hitting the same thing on nixos:
```
authReconfig: ra=false dns=true 0x01: dns set: running /nix/store/q6ylicsava582g55mdx0788yg2fib7js-systemd-246.6/bin/resolvconf -m 0 -x -a tun-tailscale.inet: Failed to resolve interface "tun-tailscale": No such device
```

(from `journalctl --unit tailscale.service`.)

**Comment by danderson** [comment]
Should be fixed in Tailscale 1.8, our DNS manager detection logic is much improved and should reliably pick resolved.
